### DISCUSSION TITLE
Ulauncher closes when mouse focus is lost regardless of setting

### DISCUSSION THREAD

**Discussion by itsyigid** [Ulauncher/Ulauncher#1268] [Q&A] [answered]
I use [Hyprland](https://github.com/hyprwm/Hyprland) on Arch Linux and recently installed Ulauncher. The experience and design are excellent, but I need help tackling the issue that Ulauncher hides when the mouse focus is lost, even though I set the option to not to.

I bound SUPER + D to command `ulauncher-toggle` and open Ulauncher by keyboard. When I move my mouse to click on entries, usually the cursor is not in center, so "the focus" on Ulauncher is "lost".

This shouldn't cause Ulauncher to hide because of the settings option, but it does.

**Comment by Llamatron2112** [comment]
You can use `windowrule = stayfocused,^(Ulauncher)$` in hyprland.conf, it will keep focus as long as it's opened

**Reply by itsyigid** [reply, marked as answer]
> You can use `windowrule = stayfocused,^(Ulauncher)$` in hyprland.conf, it will keep focus as long as it's opened

After examining ulauncher with `hyprctl clients`, I added `windowrulev2 = stayfocused,class:^(ulauncher)$` which solved my issue. Thank you!
